projects/plugins/jetpack/extensions/plugins/sharing/sharing.php

```diff
@@ -0,0 +1,40 @@
+<?php
+/**
+ * Block Editor - Sharing feature.
+ *
+ * @package automattic/jetpack
+ */
+
+namespace Automattic\Jetpack\Extensions\Sharing;
+
+use Jetpack_Gutenberg;
+
+/**
+ * Register Sharing plugin.
+ *
+ * @return void
+ */
+function register_plugins() {
+	// Register Sharing.
+	Jetpack_Gutenberg::set_extension_available( 'sharing' );
+}
+
+add_action( 'jetpack_register_gutenberg_extensions', __NAMESPACE__ . '\register_plugins' );
+
+/**
+ * The Sharing panel is only displayed for post types that support sharing.
+ * The sharing module declares support for sharing for all the public post types.
+ * Let's do the same thing when the module isn't active yet.
+ */
+add_action(
+	'init',
```

Comment:
Should we / can we hook into `rest_api_init` here, like it's done here?
https://github.com/Automattic/jetpack/blob/0f1eace8d38a3ae9d8286f27df75b37133b29bdd/projects/plugins/jetpack/modules/sharedaddy/sharing.php#L777

Reply:
Hmm, I've just thought of it. I'm not as experienced in this codebase yet. But, is that correct that when Sharing Module is disabled, this like you mentioned `add_action( 'rest_api_init', 'jetpack_post_sharing_register_rest_field' );` won't actually be run. Thus, until we enable the module, we actually don't have support for these? Maybe we shouldn't mark them as supported `add_post_type_support( $post_type, 'jetpack-sharing-buttons' );` by then? Or I'm getting it wrong?

Reply:
Oh, I think I see now why it was done in the first place. `<PostTypeSupportCheck supportKeys="{{module}}">` is basically providing a guard over rendering component. When we actually activate a module, that triggers component re-rendering , but, the module itself haven't activated everything, particularly, hasn't registered endpoints and marked `post type support`. I think ideally we would have to do `useModuleStatus` aware of actual module state and then we would be able to rely on it, and not registered `post type support` prior to Module activation itself